### PR TITLE
feat:create card model with add method

### PR DESCRIPTION
**Pull request by LeVHoangduc**
@lenguyenthimy-agilityio @hungnguyenhuu-agilityio Please help me to check and merge it.

### CODE REVIEW

File: src/javascripts/models/card.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import ApiService from "../services/apiService";
+
+class Card {
+  constructor() {
+    this.service = new ApiService("/cards");
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/models/card.js]: naming is not good. pls update

Comment (LeVHoangduc) [reply src/javascripts/models/card.js]: @hungnguyenhuu-agilityio I want to ask somthing, what naming do you mean in it?

Comment (hungnguyenhuu-agilityio) [reply src/javascripts/models/card.js]: You are initializing the API service, but you just naming the service. If you have another service which need to init, what should it name?

File: src/javascripts/models/card.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    return addCard;
+  };
+
+  /**
+   * Method to delete a card from the card list.
+   * @param {string} cardId - The ID of the card to be deleted.
+   */
+  deleteCard = async (cardId) => {
+    console.log("id of model is:", cardId);
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/models/card.js]: console.log? forgot to remove it?

**Comment by LeVHoangduc** [comment]
@hungnguyenhuu-agilityio I have updated it. Please help me to check again

File: src/javascripts/models/card.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    return addCard;
+  };
+
+  /**
+   * Method to delete a card from the card list.
+   * @param {string} cardId - The ID of the card to be deleted.
+   */
+  deleteCard = async (cardId) => {
+    await this.apiService.deleteItem(cardId);
+  };
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/models/card.js]: This function don't return anything?

File: src/javascripts/models/card.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    return data;
+  };
+
+  /**
+   * MEthod to add a new card to the card list.
+   * @param {Object} cardCurrent - The card object to be added.
+   * @returns {Promise} A promise that resolves with the result of adding the card.
+   */
+  addCard = async (cardCurrent) => {
+    const addCard = await this.apiService.postItem(cardCurrent);
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/models/card.js]: Naming of constant

File: src/javascripts/models/card.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  /**
+   * Method to retrieve the list of card from the API.
+   * @returns {Promise<Object[]>} A promise that resolves with the list of card.
+   */
+  getCardList = async () => {
+    const data = await this.apiService.getList();
+
+    return data;
```
Comment (hungnguyenhuu-agilityio) [inline src/javascripts/models/card.js]: After calling the await, and this one will still return the promise of data?

Comment (LeVHoangduc) [reply src/javascripts/models/card.js]: @hungnguyenhuu-agilityio  Yes, and promises are used for later.

File: src/javascripts/models/card.js (hunk 4 above)
Comment (LeVHoangduc) [reply src/javascripts/models/card.js]: @hungnguyenhuu-agilityio I have updated it

File: src/javascripts/models/card.js (hunk 3 above)
Comment (hungnguyenhuu-agilityio) [reply src/javascripts/models/card.js]: @LeVHoangduc You missed that

Comment (LeVHoangduc) [reply src/javascripts/models/card.js]: @hungnguyenhuu-agilityio I have updated it with returns true for this function

**Comment by LeVHoangduc** [comment]
@hungnguyenhuu-agilityio I have updated my card model. Please help me to check it again.

**Comment by LeVHoangduc** [comment]
@hungnguyenhuu-agilityio I have updated it. Please help me to check again.

**Pull request merged**